v2/tools/generator/internal/codegen/pipeline/create_types_for_backwards_compatibility.go

```diff
@@ -73,6 +76,69 @@ func CreateTypesForBackwardCompatibility(prefix string) *Stage {
 	return stage
 }
 
+func createBackwardCompatibleDefinitions(
+	resources astmodel.TypeDefinitionSet,
+	definitions astmodel.TypeDefinitionSet) (astmodel.TypeDefinitionSet, error) {
+
+	// Find all type definitions needed for the resources specified
+	defs, err := astmodel.FindConnectedDefinitions(definitions, resources)
+	if err != nil {
+		return nil, errors.Wrap(err, "finding types connected to resources requiring backward compatibility")
+	}
+
+	// Update the description of each to reflect purpose
+	withDescriptions, err := addCompatibilityComments(defs)
+	if err != nil {
+		return nil, errors.Wrap(err, "changing comments of types created for backward compatibility")
+	}
+
+	// Rename all the types into our compatibility namespace
+	renames := createBackwardCompatibilityRenameMap(defs, "v1alpha1api")
+	visitor := astmodel.NewRenamingVisitor(renames)
+	renamed, err := visitor.RenameAll(withDescriptions)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating types for backward compatibility")
+	}
+
+	return renamed, nil
+}
+
+func findResourcesRequiringCompatibilityVersion(
+	prefix string,
+	definitions astmodel.TypeDefinitionSet,
+	configuration *config.ObjectModelConfiguration) (astmodel.TypeDefinitionSet, error) {
+	compat := make(astmodel.TypeDefinitionSet)
+	var errs []error
+	resources := astmodel.FindResourceDefinitions(definitions)
+	for name, def := range resources {
+
+		// Find out when we started supporting this resource
+		from, err := configuration.LookupSupportedFrom(name)
+		if err != nil {
+			if config.IsNotConfiguredError(err) {
+				// $supportedFrom is not configured, skip this resource
+				continue
+			}
+
+			// If something else went wrong, keep details
+			errs = append(errs, err)
+		}
+
+		if strings.HasPrefix(from, prefix) {
+			compat.Add(def)
+		}
+	}
+
+	if len(errs) > 0 {
```

Comment:
Same minor error comment about being able to check `!= nil` if you make the `kerrors.NewAggregate` first

Reply:
Done.